spinoso-string/src/lib.rs

```diff
@@ -1195,18 +1195,18 @@ impl String {
     /// ```
     /// use spinoso_string::{Encoding, String};
     ///
-    /// let mut s = String::utf8(b"xyz".to_vec());
+    /// let s = String::utf8(b"xyz".to_vec());
     /// assert_eq!(s.encoding(), Encoding::Utf8);
-    /// s.make_binary();
-    /// assert_eq!(s.encoding(), Encoding::Binary);
+    /// let b = s.to_binary();
+    /// assert_eq!(b.encoding(), Encoding::Binary);
```

Comment:
Can we add another line to this doc test that asserts the byte content of the e old and new strings are the same?

Reply:
What's your favourite way to check bytes in this case? I imagine there's many ways:
- `assert_eq!(s, b)` works because the EncodedString PartialOrd is comparing the slices
- `assert_eq(*s.as_slice(), *b.as_slice())` to me is more explicit and doesn't rely on the above knowledge

Reply:
I think you're right. I like the slice way better since it doesn't rely on folks to know the partial eq behavior of `String`.

When you add this, can you remove the prefix `*`. It'll compile fine without it and it will make the test a bit clearer to read